### ISSUE TITLE
Tray Icon / Taskbar no unread message count shown

### ISSUE THREAD

**Issue by CitizenDev** [RocketChat/Rocket.Chat.Electron#1395]
### My Setup

Operating System: WIN 10 PRO 1903
App Version: 2.16.2
Server version 2.0.0
Installation type: exe

- [x] I have tested with the latest version
- [x] I can simulate the issue easily

### Description
There is never actual unread count number on tray icon (just one dot over gray icon if there are unread messages)
And no unread count on taskbar

### Current Behavior
No unread count on tray or taskbar icon

### Expected Behavior
I've read the code and at least tray icon should show numbers from 1-9 over tray icon.

It seems that the client is detecting the system as macOS not Windows and therefore shows icons from darwin directory not win32

**Comment by CitizenDev** [comment]
I now noticed that said icons are on this location \src\public\images\tray\darwin
Can someone explain why arent these used instead? \src\public\images\tray\win32

**Comment by CitizenDev** [comment]
#1376

**Comment by CitizenDev** [comment]
**Update:** Issue still persists in 2.17.1

**Comment by tassoevan** [comment]
@CitizenDev Just to be sure, is the `Unread Tray Icon Alert` setting enabled on Rocket.Chat server?

**Comment by CitizenDev** [comment]
> @CitizenDev Just to be sure, is the `Unread Tray Icon Alert` setting enabled on Rocket.Chat server?

Yes. I also tested by just using open.rocketchat server instead of my own to be sure the issue is not my server configuration.

**Comment by kosmakov** [comment]
The same problem.
The setting is not saved inside the application; after setting it to the on position, it is reset to off.
![image](https://user-images.githubusercontent.com/59031690/75055616-6ad86800-54e6-11ea-9153-b87cc6f25232.png)
Affected only Win client.

**Comment by kosmakov** [comment]
Found solution: you need to setup Require Interaction to Dismiss Desktop Notification to ON
![image](https://user-images.githubusercontent.com/59031690/75056595-68770d80-54e8-11ea-98d0-7a7d81ad99af.png)

**Comment by tassoevan** [comment]
@gabriellsh Was it an issue on Rocket.Chat's server configuration? I think we've discussed it before.

**Comment by johnyb0y** [comment]
@tassoevan the issue you linked is so old, I don’t remember the details anymore. 
All I can say is that right now with current versions we‘re disabling the „Taskbar Icon“ setting somewhere under „View“ in the client. (Not exactly show how it’s called. I’m on mobile right now and can’t check.)

That leads to the client being shown in taskbar again when minimized and I think the „orange indicator“ on new messages also works this way.

**Comment by tassoevan** [comment]
Thanks @johnyb0y. That's how it should work: hiding the window when the tray icon is enabled (true by default).

**Comment by CitizenDev** [comment]
@tassoevan The issue still persist in latest 3.0.2 version.
I have 8 unread messages and taskbar icon still does not show the number of unread messages.
This has definitely worked on versions way back.
![image](https://user-images.githubusercontent.com/23089289/95178418-13aeab80-07c8-11eb-9b63-b6cb9ed0020d.png)

**Comment by MrPoint1400** [comment]
+1

**Comment by jeanfbrito** [comment]
On MacOS there is: 
![image](https://user-images.githubusercontent.com/1177482/161975613-b5e6e3d4-775d-4ec4-a143-5ad8b69b93e2.png)
On Windows too:
![image](https://user-images.githubusercontent.com/1177482/161975924-3e3765c1-82da-4085-9316-6518228d937a.png)
The problem is that the icon is too small on the tray, that you cant see the number. I will see if there is a way to add a number like in MacOS. If not, we will need to remake the icons for it.

**Comment by CitizenDev** [comment]
For me the actual number appeared only after updating to WIN 11. So there might be issue related to WIN10 specifically.

Even though @jeanfbrito posted image where the number is clearly seen and it seems to be WIN10.

**Comment by jeanfbrito** [comment]
> For me the actual number appeared only after updating to WIN 11. So there might be issue related to WIN10 specifically.
> 
> Even though @jeanfbrito posted image where the number is clearly seen and it seems to be WIN10.

Yes, is Windows 10. I didn't like Windows 11 :)

**Comment by thestelz** [comment]
I currently don't even have the number on Mac (`12.2.1`) either with the latest Rocket.Chat version (`3.8.2`)

<img width="40" alt="Screen Shot 2022-04-07 at 10 17 10 AM" src="https://user-images.githubusercontent.com/1774502/162221147-f6535a8a-1869-4b5b-a52c-eaa5bef2f7df.png">

**Comment by thestelz** [comment]
> Do you have unread messages? Lets investigate what is happening. Give more information.

@jeanfbrito Yes, I had two unread messages. This is what the dock icon looks like:

<img width="53" alt="Screen Shot 2022-04-07 at 10 32 08 AM" src="https://user-images.githubusercontent.com/1774502/162223708-6dfa9734-e5af-4406-b62e-c8003cd41d73.png">

*Unread Tray Icon Alert* is enabled for all of my accounts

<img width="626" alt="Screen Shot 2022-04-07 at 10 34 35 AM" src="https://user-images.githubusercontent.com/1774502/162224059-49ea9743-f705-4814-8270-2755765c0921.png">

Federation is setup for cross-company communications. 

I don't mean to pollute this thread, so I can create a new issue if it doesn't pertain to this. Also, I'm not sure what the server version is as I'm not in charge of that department, but I can find out if that would also help/be useful.

**Comment by ldefabris** [comment]
I got this error on snap ubuntu 3.9.8 and now 3.9.9. Before Taskbar Tray Icon show alert about unread messages but now only sound plays and the tray icon always the same with or without unread messages...

**Comment by jeanfbrito** [comment]
> I got this error on snap ubuntu 3.9.8 and now 3.9.9. Before Taskbar Tray Icon show alert about unread messages but now only sound plays and the tray icon always the same with or without unread messages...
> 
> ![2023-10-04_14-20](https://user-images.githubusercontent.com/6388880/272658027-cc8e121d-28ac-4e98-b278-754295a19d73.png)

What server version?

**Comment by NR-THA** [comment]
We have the same Problem with 3.9.8 and now 3.9.9 on MacOS Ventura and Sonoma.
Serverversion is 5.4.10
![RocketMac](https://github.com/RocketChat/Rocket.Chat.Electron/assets/147517935/9d3d3d6a-abaa-44ca-9746-fc72cde73793)
![RocketMac 2](https://github.com/RocketChat/Rocket.Chat.Electron/assets/147517935/55181056-5b12-45e8-aff4-67f906e40800)

**Comment by ldefabris** [comment]
How I can check my server version? Don t show on about menu...

**Comment by NR-THA** [comment]
> How I can check my server version? Don t show on about menu...

I do login to the Browserversion as Admin, then wit the 3 Points to "Workspace"
![Rocket Admin](https://github.com/RocketChat/Rocket.Chat.Electron/assets/147517935/abc38994-7ed8-4ce8-90a7-45b17f2f23f2)
